### ISSUE TITLE
No way to cancel sync, had to reboot phone to stop upload

### ISSUE THREAD

**Issue by jfburke3** [mapillary/mapillary_issues#716]
This morning, I couldn't cancel sync of my weekend photos.  In the Notifications area on my phone, I had the Mapillary "sending xx of yy photos" but no Cancel sync button below it (and now that I think about it, I might not have had the Cancel button available when I was at the coffee shop yesterday, either).

* Tried disabling Upload in background.  10 minutes later it was still uploading.
* Tried disabling Upload on battery.  8 or so minutes later, it was still uploading, and my phone was NOT plugged in to the charger.
* No confidence that disabling wifi on my phone or leaving range of my home wifi would stop upload (see https://github.com/mapillary/mapillary_issues/issues/710), so I ended up rebooting my phone, which DID work, in order to be able to drive to work without uploading over cell.

Currently running Peter's special dropbox version 1.14, stock Android 4.2.2.

**Comment by peterneubauer** [comment]
Any chance you can send a bug report when that is happening?

**Comment by jfburke3** [comment]
Yes, I can do that the next time.  I'll assume the "help" address unless you specify otherwise, and reference this bug# in the subject.

**Comment by jfburke3** [comment]
Or would you rather I attach the logs here?

**Comment by peterneubauer** [comment]
Yeah,
sending the logs by mail is probably best, with a reference to the bug.

Thanks!

/peter

**Comment by jfburke3** [comment]
Logs sent.

To reproduce what I had this morning, I did the following:

* Shot a short sequence (120 pics).
* Made sure my phone was NOT plugged in to power.
* Enabled Upload in background and Upload on battery.
* Started upload.

Noted that I did not have a Cancel Sync button.

At 10 pics up, I disabled Upload in background (which I didn't expect to have an effect at this time).  Went back to the notifications pull-down to track progress.  Noted that I still didn't have a Cancel Sync button.

At 30 pics, I disabled Upload on battery.  Went back to the notifications pull-down to track progress.  Noticed that the Cancel Sync button was now present.

At 80 pics, I hit the Send debug logs button.

The pictures finished uploading while still not plugged in.  Got the "Nice job, you uploaded...." message in the notifications list.



** Special note:  From the earlier occurrance today, when I had to reboot my phone to stop the upload, I had 728 pics left to upload.  When I got to work, I connected to wifi and started the upload of that set to let it complete, which it did.  I DID NOT get the "Nice job, you uploaded...." message.  It stayed on 728 of 728 uploaded.  The logs might have data from that state as well, not sure.
